### PR TITLE
Add simple metrics of Lua usage

### PR DESCRIPTION
**Pull request by maxim-productengine**
Add two counter metrics: total scripts and auto-run scripts.

### CODE REVIEW

File: indra/newview/llluamanager.h [hunk 1]
```diff
@@ -84,6 +84,9 @@ class LLLUAmanager
 
     static const std::map<std::string, std::string> getScriptNames() { return sScriptNames; }
 
+    static S32 sAutorunScriptCount;
+    static S32 sScriptCount;
```
Comment (nat-goodspeed) [inline indra/newview/llluamanager.h:88]: Maybe a stats struct? I can think of a few additional Lua-related statistics of potential interest, even in the short term:
* the high-water mark for number of concurrently running Lua scripts (Lua-targeted C++ coroutines)
* stats based on totaling interrupts: low, median, high interrupt count per script; total Lua interrupts for the viewer session.

Lua interrupts don't correlate with real time, or even with Lua bytecode steps, but they're sort of like gauges measuring "engine hours" for heavy equipment: they indicate (in an admittedly unspecified way) the number of cycles the Lua engine spends doing script work. We might capture "kilointerrupts" if the numbers get absurdly large.

File: indra/newview/llviewerstats.cpp [hunk 2]
```diff
@@ -673,6 +674,10 @@ void send_viewer_stats(bool include_preferences)
 
     system["shader_level"] = shader_level;
 
+    LLSD &scripts = body["scripts"];
+    scripts["lua_scripts"] = LLLUAmanager::sScriptCount;
+    scripts["lua_auto_scripts"] = LLLUAmanager::sAutorunScriptCount;
```
Comment (nat-goodspeed) [inline indra/newview/llviewerstats.cpp:679]: or `body["scripts"] = llsd::map(...);`

**Review by nat-goodspeed — APPROVED** [review]

File: indra/newview/llluamanager.h (hunk 1 above)
Comment (maxim-productengine) [reply indra/newview/llluamanager.h:88]: Yep, it makes sense to use a struct if we have more than 2-3 stats.
Agree, those two may be useful. We should discuss it.
 But I can imagine that a script with an infinite loop can spoil total Lua interrupts for the viewer session.

Comment (nat-goodspeed) [reply indra/newview/llluamanager.h:88]: Okay, how about: we count interrupts for individual scripts, deferring contributing to the viewer session total until script termination. But if we terminate a script with the "infinite loop" error, we zero its interrupt statistic so it doesn't throw off stats.

Comment (maxim-productengine) [reply indra/newview/llluamanager.h:88]: Yes, it makes sense.

**Pull request merged**